test/e2e/apimachinery/garbage_collector.go

```diff
@@ -1102,10 +1102,13 @@ var _ = SIGDescribe("Garbage collector", func() {
 		}
 
 		// Wait 30s and ensure the dependent is not deleted.
-		ginkgo.By("wait for 30 seconds to see if the garbage collector mistakenly deletes the dependent crd")
-		if err := wait.PollWithContext(ctx, 5*time.Second, 30*time.Second+gcInformerResyncRetryTimeout, func(ctx context.Context) (bool, error) {
+		ginkgo.By("wait for 90 seconds to see if the garbage collector mistakenly deletes the dependent crd")
+		if err := wait.PollUntilContextTimeout(ctx, 5*time.Second, 30*time.Second+gcInformerResyncRetryTimeout, false, func(ctx context.Context) (bool, error) {
 			_, err := resourceClient.Get(ctx, dependentName, metav1.GetOptions{})
-			return false, err
+			if err != nil {
+				return false, err
+			}
+			return true, nil
```

Comment:
This doesn't look right. If it returns `true` here, `PollUntilContextTimeout` will return a nil error and the test will pass in a situation where it should have failed.

In general, using any of the `wait.Poll` variants to check that something is true consistently obscures the intent of the test because one expects an error and then has to check for the "right" error.

Can you use `gomega.Consistently` instead? It might be as simple as:
```
gomega.Consistently(ctx, framework.Get(resourceClient.Get, dependentName, metav1.GetOptions{}).WithTimeout(30*time.Second+gcInformerResyncRetryTimeout).WithPollInterval(5*time.Second).Should(gomega.Succeed())
```

Reply:
Yes, you're right. 
Thanks for your suggestions. 
Can I modify it like this? Because I didn't find the method framework.Get
```
gomega.Consistently(func() error {
  _, err := resourceClient.Get(ctx, dependentName, metav1.GetOptions{})
  return err
}, 30*time.Second+gcInformerResyncRetryTimeout, 5*time.Second).Should(gomega.BeNil())
```